rfcs/0000-buildpack-resource.md

```diff
@@ -0,0 +1,104 @@
+**Problem:**
+
+The current v1alpha1/v1alpha2 Builders and ClusterBuilders are configured with an explicit ClusterStore as the "repository" of buildpacks available to construct the order specified in `.spec.order`. The ClusterStore is a kpack resource that provides a list of buildpackage images in `.spec.sources` and is not analogous to any other concept within the CNB ecosystem.
+
+The ClusterStore being the single location for all buildpack images used in multiple builders makes managing the available buildpacks within kpack cumbersome. Adding new buildpacks requires modifying an existing resource and removing buildpacks requires carefully selecting the buildpack image to remove from the list of buildpackage images. The `kp` cli was built to handle this complexity but, the `kp` cli should not be a prerequisite to managing buildpacks within kpack.
+
+The ClusterStore being a monolithic specification of all available buildpacks leads to performance issues within kpack. If the list of available buildpacks within a ClusterStore is lengthy the reconciliation of a single ClusterStore can take considerable time. If a single buildpack within the ClusterStore is unavailable reconciliation will fail which will cause the entire contents of the ClusterStore to be unavailable.
+
+**Outcome:**
+
+Allow Buildpacks within kpack to be managed independently of other buildpacks with their own unique lifecycles with a new ClusterBuildpack/Buildpack resource while eventually removing support for the ClusterStore.
+
+**Actions to take:**
+
+Introduce a new ClusterBuildpack and Buildpack resource that provides a buildpack to be available within a Cluster or a singular namespace and be used in ClusterBuilders and Builders.
+
+The Buildpack resource will allow for an individual buildpack to be configured with a single image reference.
+
+Sample Cluster Buildpack Resource
+
+```yaml 
+apiVersion: kpack.io/v1alpha3
+kind: ClusterBuildpack
+metadata:
+  name: paketo-java-buildpack-5.9.1
```

Comment:
naming conventions are kinda important here, having version in the name would probably be standard

Reply:
Are we creating an undesired workflow where users create source of truth `Buildpack` resource named `paketo-java-buildpack` and then simply update that buildpack instead of create new ones?

Reply:
I think kpack should be un-opinionated about the workflow here. If users want to continually update a Buildpack instead of creating a new buildpack that should work!